idaes/models/properties/modular_properties/reactions/equilibrium_forms.py

```diff
@@ -189,7 +189,7 @@ def return_expression(b, rblock, r_idx, T):
         if Qunits is not None:
             Q = Q / Qunits
 
-        return s - smooth_max(0, s - Q, rblock.eps) == 0
+        return Q - smooth_max(0, Q - s / (s + b.k_eq[r_idx]), rblock.eps) == 0
```

Comment:
I have a few questions about this formulation:

1. Firstly, this looks like something that can and should be handled by scaling the variables for the solid flow so they are not so close to zero.
2. Secondly, why did you use the equilibrium constant in the normalizing term?

Reply:
`s` here is the solid molar flow rate `flow_mol_phase_comp['Sol']` and it has no clear upper bound. Depending on the feed flow rate and composition, it could end up in any order of magnitude. It's hard to properly scale it before solving the equilibrium. 

While `Q` = Ksp-f(c) has clear bounds, i.e., [0,Ksp], Ksp is typically 1e-37~1e-4 for solubility products before nondimensionlization (expect smaller Ksp). 

The original form easily ends up a positive and large `s` while `Q` is not a perfect zero considering the relative difference between f(c) and Ksp. 

So I first considered `s/(s+a)` to make the precipitate term in [0,1]. Then I chose a = Ksp to bring up similar accuracy in s and Q (or f(c)). Now the constraint can be easily scaled as other equilibrium reactions where usually something divided by their Keq.

While this `solubility_product` is still not perfect in dealing with small concentrations. We have the `log_solubility_product`, where `Q` = ln(Ksp)-ln(f(c)), with bounds [0,ln(Ksp)-ln(0)]. The upper bound ln(Ksp)-ln(1e-50) is ~1e2 considering the lower bounds of `mole_frac_phase_comp` and for common low concentrations, Q is ~1e1. This is why I consider a factor of 10 of the precipitate term in `log_solubility_product`.

To compare `s` and `Q` (decide which one is the zero term), we need similar

Reply:
See the derivatives in the following comment, a=Ksp is a very robust setup considering the sensitivity of f(c) in `Q`, and also Ksp is a constant property. Though Ksp varies with temperature (similar to Keq), it is always in a similar order of magnitude as f(c).

In general, this formulation is to refine the zero value comparison between `s` and `Q` and guarantee a similar accuracy for both terms regardless of any scaling provided.

Reply:
But `Ksp` is not constant (unless you make a simplifying assumption). My question is whether `a` should be a constant of similar magnitude to `Ksp` instead; this would help with sparsity, scaling and initialization as well as it would be `a` and `Ksp` separate.

Reply:
Maybe `k_eq_ref`? The Ksp at the reference temperature. Ksp will be larger when the temperature increases, and it will still keep accuracy in small `s`. But when the temperature goes below the reference temperature, Ksp will decrease, and we may just lose 1-2 orders of magnitude accuracy.

Reply:
I would not couple it explicitly to any physical variable - just make it a separate smoothing/scaling parameter that the user can set. `K_eq_ref` would be a good default value, but the issue again is that that is not always known a priori. Also, by uncoupling it it lets the (experienced) user pick a value that is good for their application; e.g. if they know they are notably above the conditions for the reference state, then they can estimate `K_sp` for that state and use that as the smoothing parameter.